adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1214,6 +1157,30 @@ public void startActivityForResult(Intent intent, int requestCode) {
         };
     }
 
+    private String mergeClaimsWithClientCapabilities(String claims) throws JSONException {
+        if (mClientCapabilites == null || mClientCapabilites.isEmpty()) {
+            return claims;
+        }
+        String claimsResult = null;
+        JSONArray capabilitiesArray = new JSONArray();
+
+        for (String capability : mClientCapabilites) {
+            capabilitiesArray.put(capability);
+        }
+
+        JSONObject capabilities = new JSONObject();
```

Comment:
variables in this function need to be final. also a big comment on the top on what is the formatted output will be nice.